lib/trollop.rb

```diff
@@ -47,6 +244,56 @@ class VersionNeeded < StandardError
 ## and consider calling it from within
 ## Trollop::with_standard_exception_handling.
 class Parser
+
+  ## The registry is a class-variable Hash of registered option types.
+  ## By default, it contains flag(boolean), int, float, string, io and date types
+  ## and their plural versions.
+  @registry = {
+    # single opts
+    :fixnum => IntegerOption,
+    :int => IntegerOption,
+    :integer => IntegerOption,
+    :float => FloatOption,
+    :double => FloatOption,
+    :string => StringOption,
+    :bool => BooleanOption,
+    :boolean => BooleanOption,
+    :flag => BooleanOption,
+    :io => IOOption,
+    :date => DateOption,
+    :trueclass => BooleanOption,
+    :falseclass => BooleanOption,
+    ## array-arg opts
+    :fixnums => IntegerArrayOption,
+    :ints => IntegerArrayOption,
+    :integers => IntegerArrayOption,
+    :doubles => FloatArrayOption,
+    :floats => FloatArrayOption,
+    :strings => StringArrayOption,
+    :dates => DateArrayOption,
+    :ios => IOArrayOption,
+  }
+  
+  ## Gets the class from the registry.
+  ## Can be given either a class-name, e.g. Integer, a string, e.g "integer", or a symbol, e.g :integer
+  def self.registry_getklass(type)
+    return nil unless type
```

Comment:
do we need to support `nil`?

Reply:
I think we need this.  This method gets called twice, once if the `:type` parameter is given to `opt`, and a second time if we have to deduce the option-type from the defaults.  In many cases, `:type` is not provided, and is `nil`, so then the second call will happen.  Another option to could be to null-guard using an `UnknownOption` class, perhaps?

Reply:
Keep. Thanks